### DISCUSSION TITLE
Versions incrementing quickly...

### DISCUSSION THREAD

**Discussion by JeremyTBradshaw** [microsoftgraph/msgraph-sdk-powershell#2304] [Q&A]
I'm not really experienced in developing anything in a team environment where we constantly increase the version.  My own stuff I just try to following general tips.

Wondering why the version is going 2.0.0 > 2.1.0 > 2.2.0 .... etc., rather than seeing anything like 2.0.0 > 2.0.1 > 2.0.2, ... etc. ?

Is it just that the releases are being done at the n.n.0's?  Or just not doing the 3rd position at all?  Curious/wondering, not inferring anything.

**Comment by sebastienlevert** [comment]
Hello @JeremyTBradshaw! This is a very valid question! We are using [semver](https://semver.org/) as the way to identify versions of all our client libraries on Microsoft Graph (SDKs, PowerShell, CLI, etc.). Semver is very "strict" in what is a major, minor or patch version (https://devhints.io/semver). 

* MAJOR: incompatible API changes
* MINOR: add functionality (backwards-compatible)
* PATCH: bug fixes (backwards-compatible)

So as we are adding new endpoints and properties on a weekly basis, we need to bump our minor versions weekly. Hope this helps!

**Reply by JeremyTBradshaw** [reply]
Ahhh...  thank you for that.  I will make myself remember this, new middle number increasing = adds/changes/removes whereas final number increasing= bugfixes.

Makes perfect sense.  Can imagine 0-9 will go by pretty quickly.  Follow in question is, is 10 adds/changes/removes enough to always merit a major version bump?

I think the answer is going to be: yes, or add digits to the number places (which I don't know if is even legal), e.g., 2.00.0 - 2.99.0

Thanks again, I'll also freshen up on semver directly.

**Reply by JeremyTBradshaw** [reply]
Actually good to go, I get it.  0-9 question was moot, you guys will just go from 2.9.0 to 2.10.0, nice and easy.

**Reply by sebastienlevert** [reply]
Exactly! We will bump up until we hit a major change, where we would be bumping up to 3.0.0!
